In [66]:
import math
"""def calculate_density(C, time, step_size, starting_value):
    for i in range(1, int(time/step_size)+1):
        starting_value += C*starting_value*step_size
#        print(starting_value)
    return starting_value
print(calculate_density(1.03526, 1, 0.1, 10))
exp_growth = lambda time, double, C: math.exp(math.log(time/double)*C)
#print(exp_growth(10, 10,2))"""
start = 10
double_time = 20
time = 1
C = (math.log(2)/double_time)
print(C)
x_final = 80
t_for_rep = math.log(x_final / start) / C
print(t_for_rep)
x_final = 100
t_for_rep = math.log(x_final / start) / C
print(t_for_rep)

0.034657359028
60.0
66.4385618977
100.0


Integrating $\frac{\delta x}{\delta t} = C x$ gives $x_{t} = x_0 \cdot e^{C t}$.
We know that $x_{0} = 10$ and $x_{20} = 20$, therefore: $20 = 10 \cdot e^{C \cdot 20} \Leftrightarrow 2 = e^{C \cdot 20} \Leftrightarrow log(2) = C \cdot 20 \Leftrightarrow C = \frac{ln(2)}{20} \Leftrightarrow C = 0.03466$.

Or more generally $C = \frac{log(2)}{D}$ where D is the doubling time.

To calculate when a certain increase from the original number has occured we can will in a corresponding multiple of the starting value and solve the formula for $t$: $x_{final} = x_0 \cdot e^{C \cdot t} \Leftrightarrow \frac{x_{final}}{x_0} = e^{C \cdot t} \Leftrightarrow ln(\frac{x_{final}}{x_0}) = C \cdot t \Leftrightarrow t = \frac{ln(\frac{x_{final}}{x_0})}{C}$
In the case of $x_{final} = 80$ this gives $t = 60$ and $x_{final} = 100$ results in $t = 66.4386$.